app/controllers/application_controller.rb

```diff
@@ -1,10 +1,15 @@
 class ApplicationController < ActionController::Base
   protect_from_forgery with: :exception
   before_action :configure_permitted_parameters, if: :devise_controller?
+  before_action :create_cart
 
   protected
 
   def configure_permitted_parameters
     devise_parameter_sanitizer.permit(:sign_up, keys: [:name])
   end
+
+  def create_cart
+    session[:cart_id] = ShoppingCart.create.id unless session[:cart_id]
```

Comment:
On a second thought, are you sure this does the same thing @diraulo ? I have not tested it but I feel this will not cut it